controllers/keda/clustertriggerauthentication_controller.go

```diff
@@ -50,16 +66,19 @@ func (r *ClusterTriggerAuthenticationReconciler) Reconcile(ctx context.Context,
 	err := r.Client.Get(ctx, req.NamespacedName, clusterTriggerAuthentication)
 	if err != nil {
 		if errors.IsNotFound(err) {
+			r.updateMetricsOnDelete(req.NamespacedName.String())
```

Comment:
I need to call this method in two places, to ensure that the metric is updated correctly, because authentication CRDs do not have finalizers and can be directly deleted.

Reply:
Does it make sense to have a finalizer for this?